File: lwjgl3/src/main/java/io/github/msameer0/rhythmicrush/lwjgl3/DesktopWindowController.java
package io.github.msameer0.rhythmicrush.lwjgl3;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.backends.lwjgl3.Lwjgl3Graphics;
import io.github.msameer0.rhythmicrush.window.WindowController;

public class DesktopWindowController implements WindowController {

    private static final int BASE_WIDTH  = 1280;
    private static final int BASE_HEIGHT = 720;

    @Override
    public void toggleFullscreen() {
        if (Gdx.graphics.isFullscreen()) {
            Gdx.graphics.setWindowedMode(BASE_WIDTH, BASE_HEIGHT);
        } else {
            Gdx.graphics.setFullscreenMode(Gdx.graphics.getDisplayMode());
        }
    }
}

File: lwjgl3/src/main/kotlin/io/github/msameer0/rhythmicrush/lwjgl3/window/DesktopWindowController.kt
package io.github.msameer0.rhythmicrush.lwjgl3.window

import com.badlogic.gdx.Gdx
import io.github.msameer0.rhythmicrush.window.WindowController

class DesktopWindowController : WindowController{
    companion object {
        private const val BASE_WIDTH = 1280
        private const val BASE_HEIGHT = 720
    }

    override fun toggleFullscreen() {
        if (Gdx.graphics.isFullscreen) {
            Gdx.graphics.setWindowedMode(
                BASE_WIDTH,
                BASE_HEIGHT
            )
        } else {
            Gdx.graphics.setFullscreenMode(Gdx.graphics.displayMode)
        }
    }
}
